lib/src/main/kotlin/io/hammerhead/karooext/models/KarooEvent.kt

```diff
@@ -376,6 +378,33 @@ data class OnNavigationState(
     data object Params : KarooEventParams()
 }
 
+/**
+ * Observe the routing-specific trace when navigation to destination or off route
+ *
+ * When navigation state is [OnNavigationState.NavigationState.NavigatingToDestination] this will
+ * be location from original location to the selected destination and will change if the user deviates from the
+ * suggested path.
+ *
+ * When navigation state is [OnNavigationState.NavigationState.NavigatingRoute] this will be null until the user
+ * goes off route at which point, this polyline will be the suggested way to return to the route and will change
+ * if the user deviates from the suggested return path.
+ *
+ * @see [OnNavigationState]
+ */
+@Serializable
+data class OnNavigationReroute(
```

Comment:
From a data model perspective, I think it would make more sense to include the reroute polyline as a nullable val on `NavigatingRoute` and have a polyline val on `NavigatingToDestination` and get rid of this extra event altogether. This way end consumers don't have to correlate between two different events to know what this means.

Reply:
I do think that's probably cleaner but didn't do it because it would require emitting a "state" change (with the base route) which didn't seem right. The state doesn't include on/off route designation or anything else that is outside of the rider's "initial choice of navigation".

Reply:
Depends on how you define state - in my opinion it makes sense to include the changing polyline there. This event is almost meaningless without the other one (you have no knowledge if this is a full or partial polyline) and in many cases I would think you would want to tack on the reroute polyline to the original route polyline to get a full sense of the rider's expected path.